chore: initial protocol api work (#26080)
cypress-io/cypress: guides/protocol-development.md

# Protocol Development

> Under normal circumstances, we retrieve from the cloud the code that is used to capture and communicate test data. In order to develop the code locally, developers will:

Reviewer: ```suggestion
In production, the capture code used to capture and communicate test data will be retrieved from the Cloud. However, in order to develop the capture code locally, developers will:
```

* Clone the `cypress-services` repo
  * Run `yarn`
  * Run `yarn watch` in `packages/app-capture-protocol`

> * Set `CYPRESS_LOCAL_PROTOCOL_PATH` to the path to `cypress-services/packages/app-capture-protocol/dist/index.js`
> * Clone the `cypress` repo
>   * Run `yarn`
>   * Execute `cypress run` on a project in record mode

Reviewer: ```suggestion
* Clone the `cypress` repo
  * Run `yarn`
  * Execute `CYPRESS_LOCAL_PROTOCOL_PATH=path/to/cypress-services/packages/app-capture-protocol/dist/index.js CYPRESS_INTERNAL_ENV=staging yarn cypress:run --record --key <record_key> --project <path/to/project>` on a project in record mode
```
